### PR TITLE
Feature/segmentation analyzer

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose

PR updates iresearch from upsteram.

Adds new mask analyzer
Performance improvements for iresearch internals
Velocypack is now used in iresearch so velocypack is moved to separate target

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** shell_server_aql
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -1768,13 +1768,13 @@ AqlValue Functions::NgramMatch(ExpressionContext* ctx, AstNode const&,
   std::vector<irs::bstring> attrNgrams;
   analyzerImpl->reset(attributeValue);
   while (analyzerImpl->next()) {
-    attrNgrams.push_back(token->value);
+    attrNgrams.push_back(static_cast<irs::bstring>(token->value));
```
Comment (gnusi) [inline arangod/Aql/Functions.cpp:1771]:
```suggestion
attrNgrams.emplace_back(token->value.c_str(), token->value.size()));
````

File: arangod/Aql/Functions.cpp [hunk 2]
```diff
@@ -1768,13 +1768,13 @@ AqlValue Functions::NgramMatch(ExpressionContext* ctx, AstNode const&,
...
   }
 
   std::vector<irs::bstring> targetNgrams;
   analyzerImpl->reset(targetValue);
   while (analyzerImpl->next()) {
-    targetNgrams.push_back(token->value);
+    targetNgrams.push_back(static_cast<irs::bstring>(token->value));
```
Comment (gnusi) [inline arangod/Aql/Functions.cpp:1777]: same here

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 3]
```diff
@@ -217,8 +217,8 @@ bool parse_ngram_vpack_config(const irs::string_ref& args, irs::analysis::ngram_
     return false;
   }
 
-  if (!slice.hasKey(PRESERVE_ORIGINAL_PARAM_NAME) ||
-      !slice.get(PRESERVE_ORIGINAL_PARAM_NAME).isBool()) {
+  if (!slice.hasKey(PRESERVE_ORIGINAL_PARAM_NAME.c_str()) ||
```
Comment (gnusi) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:220]: change `PRESERVE_ORIGINAL_PARAM_NAME` to `string_view`?

File: arangod/IResearch/IResearchFilterFactory.cpp [hunk 4]
```diff
@@ -3468,7 +3468,7 @@ Result fromFuncNgramMatch(
     irs::term_attribute const* token = irs::get<irs::term_attribute>(*analyzer);
     TRI_ASSERT(token);
     while (analyzer->next()) {
-      opts->ngrams.push_back(token->value);
+      opts->ngrams.push_back(irs::bstring(token->value));
```
Comment (gnusi) [inline arangod/IResearch/IResearchFilterFactory.cpp:3471]:
```suggestion
opts->ngrams.emplace_back(token->value.c_str(), token->value.size());

File: arangod/IResearch/IResearchIdentityAnalyzer.cpp [hunk 5]
```diff
@@ -22,7 +22,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 #include "IResearch/IResearchIdentityAnalyzer.h"
-#include "IResearch/IResearchVPackFormat.h"
+//#include "IResearch/IResearchVPackFormat.h"
```
Comment (gnusi) [inline arangod/IResearch/IResearchIdentityAnalyzer.cpp:25]: remove?

File: arangod/IResearch/IResearchViewSort.cpp [hunk 6]
```diff
@@ -48,7 +48,7 @@ bool parseDirectionBool(arangodb::velocypack::Slice slice, bool& direction) {
 
 bool parseDirectionString(arangodb::velocypack::Slice slice, bool& direction) {
   if (slice.isString()) {
-    std::string value = arangodb::iresearch::getStringRef(slice);
+    std::string value = static_cast<std::string>(arangodb::iresearch::getStringRef(slice));
```
Comment (gnusi) [inline arangod/IResearch/IResearchViewSort.cpp:51]:
```suggestion
std::string value = slice.copyString()
```

File: arangod/IResearch/IResearchAnalyzerFeature.cpp (hunk 3 above)
Comment (Dronplane) [reply arangod/IResearch/IResearchAnalyzerFeature.cpp:220]: velocypack::StringRef than. As it mostly used with vpack

**Pull request merged**